torch/nn/parallel/distributed.py

```diff
@@ -2240,24 +2240,26 @@ def _check_comm_hook(self, hook):
                 "Communication hook: return annotation should be torch.futures.Future[torch.Tensor].",
             )
 
-        if hook.__name__ in [
-            "bf16_compress_hook",
-            "bf16_compress_wrapper_hook",
-        ] and (
-            (torch.version.cuda is None and torch.version.hip is None)
-            or (
-                torch.version.cuda is not None
-                and int(torch.version.cuda.split(".")[0]) < 11
+        if hook.__name__ in ["bf16_compress_hook", "bf16_compress_wrapper_hook"]:
```

Comment:
I think you changed the original logic here. For example, in original logic, if `dist.is_nccl_available` is `False`, `_log_and_throw` will be triggered no matter if `hook.__name__` is under the condition. But your change makes all condition check is protected under condition `hook.__name__`

Reply:
I see. You are right!